README.md

```diff
@@ -217,9 +217,23 @@ Benchee.run(%{
 
 ```
 
-### Setup and teardown
+### Hooks (Setup, Teardown etc.)
 
-If you want to do setup and teardown, i.e. do something before or after the benchmarking suite executes this is very easy in benchee. As benchee is just plain old functions just do it before/after you call benchee:
+If you want to do something before or after something happens in benchee, we got you coverd! In general, there are 3 distrinct times where you might want to do something before/after it happened:
+
+* the whole suite (aka before everything/after everything) - done by just running code before/after benchee
+* an individual benchmarking scenario (one result reported you, combination of a benchmarking function and possibly an input) - `before_scenario`/`after_scenario`
+* each individual invocation of a benchmarking function - `before_each`/`after_each`
+
+benchee covers all these cases (well the first one doesn't need any specific support) and each of them will be discussed here followed by a little overview when what actually happens.
```

Comment:
I'm not entirely sure how to fix this sentence, but it reads a little weird. Maybe it should end with `followed by a little overview of what actually happens`?

Reply:
I would remove this sentence altogether.